src/adu-shell/src/main.cpp

```diff
@@ -350,9 +374,14 @@ int main(int argc, char** argv)
 
         ADUC_Logging_Uninit();
 
-        return ret;
+        goto done;
     }
 
     Log_Error("Cannot set user identity. (code: %d, errno: %d)", ret, errno);
+done:
+    if (config != NULL)
+    {
+        ADUC_ConfigInfo_ReleaseInstance(config);
```

Comment:
nit: there's a few places here where we do a NULL check before release. Might be good to make the ReleaseInstance() function handle NULL so we have an easier escape path all around.